walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -359,4 +359,30 @@ public void optionalJustEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void onlyOptionalAttendees() {
+    // No mandatory attendees, just two optional attendees with several gaps in their schedules.
+    //
+    // Events  :    |--A--|    |--B--|
+    // Day     : |---------------------|
+    // Options : |--|     |----|
+    Collection<Event> events =
+        Arrays.asList(new Event("Event 1", TimeRange.fromStartEnd(TIME_0900AM, TIME_0930AM, false),
+                          Arrays.asList(PERSON_A)),
+            new Event("Event 2", TimeRange.fromStartEnd(TIME_1000AM, TIME_1100AM, false),
+                Arrays.asList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TIME_1000AM, false),
+            TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
```

Comment:
This can be better written using: https://truth.dev/api/1.1/com/google/common/truth/IterableSubject.html#containsExactlyElementsIn(java.lang.Object%5B%5D)

This assumes you return the results in a certain order, which was not a requirement (and thus only an implementation detail).

Reply:
Sorry could you please elaborate how lines 381-386 can be written better with containsExactlyElementsIn.

Reply:
Done. Sorry took a bit longer because I was getting it confused with another Assert package.